docs/DeveloperGuide.md

```diff
@@ -378,234 +375,353 @@ Use case ends.
-* **Use Case**: UC06 - Add an Order under a Customer
+* 1a. SellSavvy cannot find any customers with at least one matching keyword.
+  * 1a1. SellSavvy tells the user that no related customers are found.
+
+  Use case ends.
+
+**Use case 5: Edit a Customer's Details**
+
+* **Use Case**: UC05 - Edit a Customer's Details
 * **Preconditions**: There are customers displayed in the customer list.
+* **Guarantees**:
+  * The specified customer's details will be overwritten if the input parameters are valid.
+
+**MSS**
+
+1. User finds the index of the customer they want to edit.
+2. User specifies the customer index along with modifications they want to make to the customer's details.
+3. SellSavvy updates the customer list with the modifications made to the customer.
+4. SellSavvy confirms the modification by stating the updated customer's details.
+
+Use case ends.
+
+**Extensions**
+
+* 2a. The customer index is missing or non-positive.
+  * 2a1. SellSavvy displays an error about the format and states the command format.
+
+  Use case ends.
+
+
+* 2b. SellSavvy detects that there are no customers with the specified index.
+    * 2b1. SellSavvy displays an error that the customer index is invalid.
+
+  Use case ends.
+
+
+* 2c. There are no customer details specified for modification.
+  * 2c1. SellSavvy tells the user that at least one field has to be edited.
+
+  Use case ends.
+
+
+* 2d. SellSavvy detects that the updated fields do not satisfy its constraints.
+    * 2d1. SellSavvy states the constraints of the invalid parameter.
+
+  Use case ends
+
+
+* 2e. SellSavvy detects that a customer with identical name already exists.
+    * 2e1. SellSavvy displays an error that the customer already exists.
+
+  Use case ends.
+
+
+* 3a. SellSavvy detects that the new name of the customer is similar to that of an existing customer.
+    * 3a1. SellSavvy gives a warning that a customer with a similar name already exists.
+
+  Use case resumes from step 4.
+
+
+* 3b. SellSavvy detects that there are tags with similar names among the updated tags.
+    * 3b1. SellSavvy gives a warning that there are similar tags on the customer.
+
+  Use case resumes from step 4.
+
+**Use case 6: Add an Order under a Customer**
+
+* **Use Case**: UC06 - Add an Order under a Customer
+* **Preconditions**: There are orders displayed in the customer's list.
 * **Guarantees**:
     * A new pending order will be added under the specified customer, if input parameters are valid.
 
 **MSS**
 
-1. User wants to add an order under a specific customer.
-2. User finds the customer index from the list.
-3. User adds the customer using the index, specifying the details of the order.
-4. SellSavvy adds the order under the customer.
-5. SellSavvy confirms the addition by displaying the newly added order's details and customer's list of orders.
+1. User finds the customer index from the list.
+2. User adds the order by specifying the customer using the index and details of the order.
```

Comment:
using the customer index and details of the order?

Reply:
will change